server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -484,7 +486,13 @@ def annotation_put_fbs(self, axis, fbs):
         # so treat this as a critical section.
         with self.label_lock:
             self.labels = new_label_df
-            write_labels(fname, self.labels)
+            lastmod = self.data_locator.lastmodtime()
```

Comment:
[optional] nit: can we round off the microseconds? assuming we won't get them out of order because the file names are numbered, it's a bit friendlier to round.

Reply:
will do.